zyte_spider_templates/spiders/serp.py

```diff
@@ -48,6 +59,61 @@ class SerpMaxPagesParam(BaseModel):
     )
 
 
+@document_enum
+class SerpItemType(str, Enum):
+    article: str = "article"
+    """
+    Article data.
+    """
+
+    articleList: str = "articleList"
+    """
+    Article list data.
+    """
+
+    forumThread: str = "forumThread"
+    """
+    Forum thread data.
+    """
+
+    jobPosting: str = "jobPosting"
+    """
+    Job posting data.
+    """
+
+    product: str = "product"
+    """
+    Product data.
+    """
+
+    productList: str = "productList"
+    """
+    Product list data.
+    """
+
+
+ITEM_TYPE_CLASSES = {
+    SerpItemType.article: Article,
+    SerpItemType.articleList: ArticleList,
+    SerpItemType.forumThread: ForumThread,
+    SerpItemType.jobPosting: JobPosting,
+    SerpItemType.product: Product,
+    SerpItemType.productList: ProductList,
+}
+
+
+class SerpItemTypeParam(BaseModel):
+    item_type: Optional[SerpItemType] = Field(
+        title="Item type",
+        description=(
+            "If specified, result URLs are followed to extract the specified "
```

Comment:
I wonder if we should make the description Google-specific. For example, there are organic and sponsored results; if I get it right, that's organic results which are followed, not sponsored - we should be explicit about it (unless we consider it a bug that sponsored results are not followed - which I'm not sure is a bug).

Reply:
Something like "Follow the links in the organic search results, and extract the selected data type from the target pages"?

Reply:
953091b?